File: struct-type/07_decorator/java/Main.java
package main;

public class Main {
 
    public static void main(String[] args) {
        System.out.println("============= 原始的正方形绘制类 ==============");
        Draw square = new Square();
        System.out.println(square.draw());

        System.out.println("============= 装饰后的正方形绘制类 ==============");
        Draw squareWrapper = new ColorSquareWrapper(square);
        System.out.println(squareWrapper.draw());
    }

}

File: struct-type/07_decorator/go/main.go
package main

import "fmt"

func main() {

	fmt.Println("============== 功能有限的原始类 ================")
	square := &Square{}
	fmt.Println(square.Draw())

	fmt.Println("============== 使用装饰器模式扩展 ================")
	redSquare := NewColorDecorator(square, "red")
	fmt.Println(redSquare.Draw())
}
